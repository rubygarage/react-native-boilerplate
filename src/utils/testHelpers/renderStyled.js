import React from 'react';
import { ThemeProvider } from 'styled-components/native';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';

import theme from 'constants/theme';

const renderStyled = (tree) => renderer.create(
  <ThemeProvider theme={theme}>
    {tree}
  </ThemeProvider>,
);

export default renderStyled;
