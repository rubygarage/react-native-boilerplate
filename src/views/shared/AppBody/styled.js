import styled from 'styled-components/native';

const AppBody = styled.View`
  position: relative;
  flex-grow: 1;
  padding-right: 16;
  padding-left: 16;
  background-color: ${({ theme }) => theme.colors.black};
`;

export default AppBody;
