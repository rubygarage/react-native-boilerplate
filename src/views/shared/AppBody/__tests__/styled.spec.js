import React from 'react';
import { View } from 'react-native';

import renderStyled from 'utils/testHelpers/renderStyled';
import AppBody from '../styled';

describe('AppBody styled component', () => {
  it('renders corectly default', () => {
    const wrapper = renderStyled(<AppBody><View /></AppBody>);
    expect(wrapper).toMatchSnapshot();
  });
});
