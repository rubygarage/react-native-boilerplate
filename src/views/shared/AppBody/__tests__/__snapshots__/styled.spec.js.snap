// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBody styled component renders corectly default 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#000000",
        "flexGrow": 1,
        "paddingLeft": 16,
        "paddingRight": 16,
        "position": "relative",
      },
    ]
  }
>
  <View />
</View>
`;
